### DISCUSSION TITLE
Windows PowerShell 5.1 "No initialization script available for powershell"

### DISCUSSION THREAD

**Discussion by sharpninja** [JanDeDobbeleer/oh-my-posh#1350] [Troubleshoot]
Init string:

```powershell
try {
    if($PSVersionTable.PSEdition -ieq "core") {
        $powershell = "pwsh"
    } else {
        $powershell = "powershell"
    }

    if(-not (get-command oh-my-posh)) {
        . scoop install oh-my-posh3
    }

    oh-my-posh --init --shell $powershell --config $env:scoop\apps\oh-my-posh3\current\themes\ys.omp.json `
        | Invoke-Expression
}
catch {
    Write-Host $Error    
}
```

The path (` $env:scoop\apps\oh-my-posh3\current\themes\ys.omp.json`) resolves correctly.  When run from Windows Powershell I receive: 

```
No initialization script available for powershell
```

**Comment by JanDeDobbeleer** [comment]
Replace `$powershell` with `pwsh`as stated in the [docs](https://ohmyposh.dev/docs/windows#replace-your-existing-prompt).

**Reply by JanDeDobbeleer** [reply]
There's no difference in invocation so I have no idea where this strategy comes from. You can't use `"powershell"` as a value for `--shell` on init

**Reply by JanDeDobbeleer** [reply]
Additionally `scoop install oh-my-posh3` isn't officially supported.

**Comment by sharpninja** [comment]
OK, so when you use pwsh for --shell, but you're in Windows PowerShell, which one am I issuing commands to at the prompt?

**Reply by JanDeDobbeleer** [reply]
I don't understand the question. pwsh is Windows Powershell. It doesn't matter if it's 5 or Core from oh-my-posh' perspective.

**Reply by sharpninja** [reply]
No, pwsh.exe is PowerShell Core and runs cross-platform on dotnet core (Now .Net 6.0 for PowerShell 7.2).  Windows PowerShell runs only on .Net Framework using the latest .Net Framework installed on the system, currently .Net Framework 4.8 on most machines.  The Windows PowerShell has access to parts of Windows that PS 7 does not.  Because of this, Windows PowerShell and PowerShell 7 coexist to handle different workloads.  PS 7 is used by Azure for almost everything.  Windows PowerShell is used almost exclusively by Windows Administrators as there are many features of Windows that cannot be manipulated from PS 7.  So it becomes vital to know which version of PowerShell is used to execute a command.  If OH-MY-POSH executes commands in PS 7, even when launch from Windows PowerShell, then I need to know that ahead of time, not when scripts are failing.

**Reply by JanDeDobbeleer** [reply]
I'm afraid you're confused as to what oh-my-posh is and does. We do not define your runtime, you start oh-my-posh in either Powershell 5 or Core, it only renders the prompt. It does not load Powershell (or any other shell) under the hood. The logic we use is the same in 5 or Core, but oh-my-posh does need to understand where it's running (hence the `--shell` flag), all it does is load the right logic so that we can hijack the prompt function (or however the shell works). You can use `--shell pwsh` in 5 or Core, it will not impact or switch the shell you opened and they both are identical when it comes to how the prompt works.

There is something to be said about us indicating the actual shell (that you're on 5 and not core or vice versa), but that's all.

**Reply by sharpninja** [reply]
Thank you for clarifying.  Showing both shell and OS would be great.

**Reply by JanDeDobbeleer** [reply]
https://github.com/JanDeDobbeleer/oh-my-posh/pull/1359
